src/offchain-ipfs-callables.md

```diff
@@ -1,14 +1,175 @@
-- api.tx.templateModule.ipfsDhtFindProviders - search for PeerIds known to be providing the given Cid; the off-chain worker interval for this activity is every block. Try it with the hash of the IPFS CID Inspector, QmY7Yh4UquoXHLPFo2XbhXkhBvFoPwmQUSa92pxnxjQuPU. Make sure you’re connected to at least one peer though!
-- api.tx.templateModule.ipfsInsertPin - pin a block with the specified Cid, making it persistent; a pinned block can’t be removed. Try pinning the 1234 CID above: QmU1f6ngsoHvwtViihzLQPXCA8j3sagmvY9GJJDY7Ao7Aa
-- api.tx.templateModule.ipfsRemovePin - remove a pin from a block, i.e. unpin it, so that it is no longer persistent and can be removed too. Trey removing the pin from the previous item.
-- api.tx.templateModule.ipfsRemoveBlock - remove a block from the node’s repository
+# Example pallet callable reference
+
+The following callable functions are exposed via JSON-RPC by the example pallet.
+These specicif functions were chosen due to popularity to give you a familiar experience.
+
+However, as we explain at the bottom of this page, these functions are not the only functions
+available in the native runtime, and therefore do not represent the full extent of functionality
+available to you.
+
+Regardless, we still detail the example template functions here as a helpful reference. If you have
+feedback about these existing functions, or want to request new functions, please open an issue at
+[rs-ipfs/substrate].
+
+[rs-ipfs/substrate]: https://github.com/rs-ipfs/substrate
+
+## Callables
+
+What follows is a list of callables, their frequency in terms of block creation, their weights,
+their arguments and return values. As explained in the previous section on the example pallet,
+weights are essentially picosecond representations of a time limit for a given transaction,
+and can be loosely correlated to transaction fees.
+
+> **Rust's `snake_case` is used here.** However, as you'll see in the upcoming polkadot.js example,
+JavaScript will use `camelCase` for function and variable names. Generally, the exposed JSON-RPC
+functions will adhere to the conventions of the programming language that the client code is
+written in.
+
+Also, while we list "return" values for simplicity's sake, what this means in the context of
+Substrate is that the values will be _eventually_ emitted returned as chain events, which you
+will need to listen for.
+
+### `ipfs_add_bytes`
+
+Adds the given bytes to the IPFS repository
+
+**Frequency:** Every other block<br />
+**Weight**: 200,000<br />
+
+#### Arguments
+
+- `bytes` - Byte array or byte string e.g. `vec![1, 2, 3, 4]` or `b"1234"`
+
+#### Returns
+
+A Content ID (CID) string, e.g. `QmU1f6ngsoHvwtViihzLQPXCA8j3sagmvY9GJJDY7Ao7Aa`
+
+### `ipfs_cat_bytes`
+
+Displays the bytes behind a given CID
+
+**Frequency:** Every other block<br />
+**Weight**: 100,000
+
+#### Arguments
+
+- `cid`: A CID string, e.g. `QmU1f6ngsoHvwtViihzLQPXCA8j3sagmvY9GJJDY7Ao7Aa`
```

Comment:
again, substrate-wise this is a `Vec<u8>` (due to wasm limitations, so we should probably note that this is from the web UI or JS perspective)

Reply:
Updated, see above question